### ISSUE TITLE
Introduce non case sensitive matching for cy.contains()

### ISSUE THREAD

**Issue by James-E-Adams** [cypress-io/cypress#2785]
### Current behavior:
`cy.contains('Capital Sentence')` does not match

`<span class="capitalize"> capital sentence </span>`

where:
```css
.capitalize {
   text-transform: capitalize;
}
```

Is this expected? 

### Desired behavior:

I assumed it would match because that's what the user sees. However the underlying text in the DOM is indeed `capital sentence`.

### Versions

Cypress: 3.0.1
Chrome, Mac OS Mojave

**Comment by brian-mann** [comment]
It would be the same problem in regards to things like `uppercase` or `truncate` for that matter. 

If you know of any DOM methods that return the text with this kind of CSS styling factored in, let us know.

**Comment by jennifer-shehane** [comment]
Introducing this into the default cy.contains() would also make it impossible to then test actually capitalization within an application. If my team wanted our content to move from Capitalized to Sentence Case, for example, there wouldn't be a way to test that those changes were put in correctly. 

If this were even implemented, I would recommend it as a separate argument to be passed like in 'search' in your IDE - to `matchCase` or `noMatchCase`.

**Comment by jennifer-shehane** [comment]
I actually might prefer to have `matchCase` as the arg and no match case as the default because I get pretty annoyed myself having to update my tests when I only changed capitalization in my app.

**Comment by vedmant** [comment]
I'd also need case insensitive contains method, maybe add some options to it?

**Comment by Bahaa-Addin** [comment]
I solved it with a regex with `i` flag for case insensitive :
`cy.contains(/capital sentence/i)`

**Comment by vesper8** [comment]
please add an argument to make cy.contains optionally work in a non case-sensitive way

using the regex works but when combined with variables requires you to add a lot of extra code.. it would be much better to have it built-in

**Comment by vesper8** [comment]
my solution is to create a helpers class and then import it inside my tests like this:

helpers.js
```
export default class Helpers {
  static caseInsensitive(str) {
    // escape special characters
    this.input = str.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
    return new RegExp(`${this.input}`, 'i');
  }
}
```

spec.js
```
import helpers from '../support/helpers';
...
cy.get('[data-cy="form-errors"]').contains(helpers.caseInsensitive(variables.strings.en.general.oops));
```

**Comment by DavidGWheeler** [comment]
> I solved it using a regex with `i` flag for case insensitive :
> `cy.contains(/capital sentence/i)`

`cy.contains()` isn't really relevant here, I think.  People are looking for a way to do a case insensitive assert. `.contains()` is like `.get()` for selecting an element which contains some text or other value.

**Comment by jennifer-shehane** [comment]
Released in [`4.0.0`](https://on.cypress.io/changelog#4-0-0).

This comment thread has been locked. If you are still experiencing this issue after upgrading to
Cypress v4.0.0, please open a new issue.
